pkg/network/dhcp/masquerade_test.go

```diff
@@ -58,22 +58,29 @@ var _ = Describe("Masquerade DHCP configurator", func() {
 		var vmiSpecNetwork *v1.Network
 		var vmiSpecIface *v1.Interface
 
-		generateExpectedConfig := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
+		generateExpectedConfigIPv6Disabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
 			ipv4Gateway, ipv4, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv4)
-			ipv6Gateway, ipv6, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv6)
 			expectedConfig := cache.DHCPConfig{Name: ifaceName,
-				IP:                  *ipv4,
-				IPv6:                *ipv6,
-				Mtu:                 uint16(mtu),
-				AdvertisingIPAddr:   ipv4Gateway.IP.To4(),
-				AdvertisingIPv6Addr: ipv6Gateway.IP.To16(),
-				Gateway:             ipv4Gateway.IP.To4(),
+				IP:                *ipv4,
+				Mtu:               uint16(mtu),
+				AdvertisingIPAddr: ipv4Gateway.IP.To4(),
+				Gateway:           ipv4Gateway.IP.To4(),
 			}
 
 			if macString != nil {
 				mac, _ := net.ParseMAC(*macString)
 				expectedConfig.MAC = mac
 			}
+
+			return expectedConfig
+		}
+
+		generateExpectedConfigIPv6Enabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
+			expectedConfig := generateExpectedConfigIPv6Disabled(vmiSpecNetwork, macString, mtu, ifaceName)
+			ipv6Gateway, ipv6, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv6)
```

Comment:
Done.
Since this changes stuff that were here before my PR, I pushed it in a different commit.
https://github.com/kubevirt/kubevirt/pull/6516/commits/9b371b17d48e9dd7b4299ae0fe45d3a239cb7da0